website/programs.py

```diff
@@ -270,6 +270,31 @@ def submit_program(self, user):
         }
         return jsonify(response)
 
+    @route("/unsubmit", methods=["POST"])
+    @requires_login
```

Comment:
only teachers should be able to unsubmit programs right? Perhaps we need to discuss this later today!